File: 02-Recursion Level-EasyMedium/03-SumOfElementsOfArray/sumOfAllElementsArray.cpp
#include<iostream>
using namespace std;

int sum(int arr[], int n) {
    if(n == 1) return arr[0];
    return arr[n-1] + sum(arr , n-1); 
}   

int main() {
    int arr[] = {1,2,3,4,5};
    int n = sizeof(arr)/sizeof(arr[0]);
    cout << "Sum of all elements in array: " << sum(arr, n) << endl;

    return 0;
}

File: 02-Recursion Level-EasyMedium/03-SumOfElementsOfArray/sumOfAllElementsArray.js
//METHOD 1 : sending array and length 
function sumOfAllElementsOfArrayWithLength(arr , len) {
    let index = len - 1; 
    if(index == 0) return arr[index];

    return arr[index] + sumOfAllElementsOfArrayWithLength(arr, len-1);
}


function sumOfAllElementsOfArrayWithIndex(arr, index) {
    if(index == 0) return arr[0];
    return arr[index] + sumOfAllElementsOfArrayWithIndex(arr , index-1);
}



let array = [1,2,3,4,5];
let len = array.length; 
let lastIndex = len - 1; 
let sum1 = sumOfAllElementsOfArrayWithLength(array , len);
let sum2 = sumOfAllElementsOfArrayWithIndex(array , lastIndex);
console.log("Sum 1 :",sum1); 
console.log("Sum 2 :",sum2); 

